backend/src/main/java/hu/flowacademy/vajdasagbrand/repository/ItemRepositoryImpl.java

```diff
@@ -0,0 +1,41 @@
+package hu.flowacademy.vajdasagbrand.repository;
+
+import hu.flowacademy.vajdasagbrand.persistence.entity.Item;
+import hu.flowacademy.vajdasagbrand.persistence.repository.ItemRepository;
+import hu.flowacademy.vajdasagbrand.dto.ItemDTO;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Repository
+public class ItemRepositoryImpl implements CommonItemRepository{
+
+    private final ItemRepository itemRepository;
+
+    public ItemRepositoryImpl(@Lazy ItemRepository itemRepository) {
```

Comment:
Nem fog kelleni a `@Lazy` mert most biztosan létezni fog a függőség. A konstruktor sem kell, szóval használj hozzá lombokot.

Reply:
ez is megvan